client.go

```diff
@@ -571,10 +571,13 @@ func (client *client) update(data *MetadataResponse) ([]string, error) {
 				toRetry[topic.Name] = true
 			}
 		}
-		var partitionCache [maxPartitionIndex][]int32
-		partitionCache[allPartitions] = client.setPartitionCache(topic.Name, allPartitions)
-		partitionCache[writablePartitions] = client.setPartitionCache(topic.Name, writablePartitions)
-		client.cachedPartitionsResults[topic.Name] = partitionCache
+
+		if topic.Err != ErrUnknownTopicOrPartition && topic.Err != ErrInvalidTopic {
```

Comment:
I think I prefer to just unconditionally delete the cache at the top of the loop for all partitions, and let it get filled in again for those we care about